samples/api/hello_triangle_1_3/hello_triangle_1_3.cpp

```diff
@@ -0,0 +1,1340 @@
+	}
+	else if (message_severity & VK_DEBUG_UTILS_MESSAGE_SEVERITY_INFO_BIT_EXT)
+	{
+		LOGI("{} Validation Layer: Information: {}: {}", callback_data->messageIdNumber, callback_data->pMessageIdName, callback_data->pMessage);
+	}
+	else if (message_severity & VK_DEBUG_UTILS_MESSAGE_TYPE_PERFORMANCE_BIT_EXT)
+	{
+		LOGI("{} Validation Layer: Performance warning: {}: {}", callback_data->messageIdNumber, callback_data->pMessageIdName, callback_data->pMessage);
+	}
+	else if (message_severity & VK_DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT)
+	{
+		LOGD("{} Validation Layer: Verbose: {}: {}", callback_data->messageIdNumber, callback_data->pMessageIdName, callback_data->pMessage);
+	}
+	return VK_FALSE;
+}
+#endif
+
+/**
+ * @brief Validates a list of required extensions, comparing it with the available ones.
+ *
+ * @param required A vector containing required extension names.
+ * @param available A VkExtensionProperties object containing available extensions.
+ * @return true if all required extensions are available
+ * @return false otherwise
+ */
+bool HelloTriangleV13::validate_extensions(const std::vector<const char *>          &required,
+                                           const std::vector<VkExtensionProperties> &available)
+{
+	bool all_found = true;
+
+	for (const auto *extension_name : required)
+	{
+		bool found = false;
+		for (const auto &available_extension : available)
+		{
+			if (strcmp(available_extension.extensionName, extension_name) == 0)
+			{
+				found = true;
+				break;
+			}
+		}
+
+		if (!found)
+		{
+			// Output an error message for the missing extension
+			LOGE("Error: Required extension not found: {}", extension_name);
+			all_found = false;
+		}
+	}
+
+	return all_found;
+}
+
+/**
+ * @brief Validates a list of required layers, comparing it with the available ones.
+ *
+ * @param required A vector containing required layer names.
+ * @param available A VkLayerProperties object containing available layers.
+ * @return true if all required extensions are available
+ * @return false otherwise
+ */
+bool HelloTriangleV13::validate_layers(const std::vector<const char *>      &required,
+                                       const std::vector<VkLayerProperties> &available)
+{
+	bool all_found = true;
+
+	for (const auto *layer_name : required)
+	{
+		bool found = false;
+		for (const auto &available_layer : available)
+		{
+			if (strcmp(available_layer.layerName, layer_name) == 0)
+			{
+				found = true;
+				break;
+			}
+		}
+
+		if (!found)
+		{
+			LOGE("Error: Required layer not found: {}", layer_name);
+			all_found = false;
+		}
+	}
+
+	return all_found;
+}
+
+/**
+ * @brief Find the vulkan shader stage for a given a string.
+ *
+ * @param ext A string containing the shader stage name.
+ * @return VkShaderStageFlagBits The shader stage mapping from the given string, VK_SHADER_STAGE_VERTEX_BIT otherwise.
+ */
+VkShaderStageFlagBits HelloTriangleV13::find_shader_stage(const std::string &ext)
```

Comment:
This could be simplified or removed all along and making the sample pass the stage to the shader loading function. The sample only uses vert and frag anyway.